src/components/block-button.js

```diff
@@ -0,0 +1,31 @@
+window.blockUser = function(id) {
+  NAF.connection.adapter.publisher.handle.sendMessage({ kind: "block", whom: id });
+  NAF.connection.entities.removeEntitiesOfClient(id);
+};
+
+window.unblockUser = function(id) {
+  NAF.connection.adapter.publisher.handle.sendMessage({ kind: "unblock", whom: id });
+  window.setTimeout(() => {
+    // Wait enough time for the unblock to be processed
+    NAF.connection.entities.completeSync(id);
+  }, 1000);
+};
+
+AFRAME.registerComponent("block-button", {
+  init() {
+    this.el.addEventListener("click", this.onClick);
+    const owner = (function getOwner(el) {
+      if (el.firstUpdateData) {
+        return el.firstUpdateData.owner;
+      }
+      if (el.parentEl) {
+        return getOwner(el.parentEl);
```

Comment:
This code goes away by using the NAF utility anway, but i'd be careful using recursion for walking up the tree like this cuz you could end up blowing up the stack. I know in es6 some sort of tail call optimizaiton was added but not too sure how/if it works with babel.

Reply:
noted. I don't want to blow up the stack